### PR TITLE
signal: Remove configurable assignment of signal numbers

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary

And replace NuttX special SIGWORK with SIGIO.
Please reference the issue here for more information: https://github.com/apache/nuttx/issues/8898

## Impact

Kconfig change only

## Testing

need https://github.com/apache/nuttx-apps/pull/1683 to pass CI.

### CODE REVIEW

File: sched/paging/pg_miss.c [hunk 1]
```diff
@@ -179,7 +179,7 @@ void pg_miss(void)
   if (!g_pftcb)
     {
       pginfo("Signaling worker. PID: %d\n", g_pgworker);
-      nxsig_kill(g_pgworker, SIGWORK);
+      nxsig_kill(g_pgworker, SIGIO);
```
Comment (pkarashchenko) [inline sched/paging/pg_miss.c:182]: This should be realtime signal.
Let's add `#define SIGWORK SIGRTMIN` into `sched/paging/paging.h` and use it

Comment (pkarashchenko) [reply sched/paging/pg_miss.c:182]: The `SIGIO` has `nxsig_abnormal_termination` signal action

Comment (pkarashchenko) [reply sched/paging/pg_miss.c:182]: Or maybe we can add call sequence similar to `sighold` for kernel task. Something like
```
sigemptyset(&set);
sigaddset(&set, SIGIO);
sigprocmask(SIG_SETMASK, &set, NULL);
```
For page worker

Comment (pkarashchenko) [reply sched/paging/pg_miss.c:182]: Or change `nxthread_setup_scheduler()` and modify line `tcb->sigprocmask = rtcb->sigprocmask;` to become
```
if (ttype != TCB_FLAG_TTYPE_KERNEL)
  {
    tcb->sigprocmask = rtcb->sigprocmask;
  }
else
  {
    tcb->sigprocmask = ALL_SIGNAL_SET;
  }
```
Like, mask all signals for kernel threads. Does this makes sense?

Comment (pkarashchenko) [reply sched/paging/pg_miss.c:182]: I mean that I do not expect that user can kill kernel thread at all.

Comment (xiaoxiang781216) [reply sched/paging/pg_miss.c]: Ok, change to SIGPAGING(SIGRTMIN).

File: sched/paging/paging.h [hunk 2]
```diff
@@ -51,6 +51,8 @@
 #  define CONFIG_PAGING_STACKSIZE  CONFIG_IDLETHREAD_STACKSIZE
 #endif
 
+#define SIGPAGING SIGRTMIN
```
Comment (pkarashchenko) [inline sched/paging/paging.h:54]: Please add include of signal.h

Comment (xiaoxiang781216) [reply sched/paging/paging.h:56]: Done.

**Review by pkarashchenko — APPROVED** [review]

**Review by patacongo — APPROVED** [review]
LGTM

**Comment by pkarashchenko** [comment]
We need to remove
```
#ifndef CONFIG_SIG_INT
#  define SIGINT 10
#else
#  define SIGINT CONFIG_SIG_INT
#endif
```
in `apps/system/cu/cu_main.c`

**Comment by xiaoxiang781216** [comment]
@pkarashchenko it's here: https://github.com/apache/nuttx-apps/pull/1683

**Comment by patacongo** [comment]
Sorry it took me so long to merge this.  I thought it was waiting for clean CI result.  I just noticed your comment.

**Pull request merged**